met-web/src/components/common/Modals/FeedbackModal.tsx

```diff
@@ -0,0 +1,342 @@
+    };
+
+    function IconContainer(props: IconContainerProps) {
+        const { value, ...other } = props;
+        return <span {...other}>{customIcons[value].icon}</span>;
+    }
+
+    function getLabelText() {
+        if (rating == 0) return;
+        return `${customIcons[rating].label}`;
+    }
+
+    function getPlaceholderText() {
+        switch (commentType) {
+            case 'Issue':
+                return 'Something does not work...';
+            case 'Idea':
+                return 'I would like to see...';
+            case 'Else':
+                return 'I was wondering...';
+        }
+
+        return '';
+    }
+
+    function handleRatingChanged(value: number) {
+        setFeedbackFormData({
+            ...feedbackFormData,
+            rating: rating == value ? 0 : value,
+        });
+    }
+
+    function handleCommentTypeChanged(value: string) {
+        setFeedbackFormData({
+            ...feedbackFormData,
+            commentType: commentType == value ? '' : value,
+        });
+    }
+
+    function handleCommentChanged(value: string) {
+        setFeedbackFormData({
+            ...feedbackFormData,
+            comment: value,
+        });
+    }
+
+    function handleSubmit() {
+        console.log(feedbackFormData);
+        setIsSubmitted(true);
+    }
+
+    function handleClose() {
+        setIsSubmitted(false);
+        setFeedbackFormData(createDefaultFeedback());
+        setIsOpen(false);
+    }
+
+    const StyledButton = styled(Button)(() => ({
+        borderColor: BaseTheme.palette.divider,
+        color: BaseTheme.palette.text.primary,
+        borderWidth: 1,
+        borderStyle: 'solid',
+        borderRadius: 5,
+        padding: 3,
+        paddingBottom: 5,
+        marginRight: 2,
+        marginLeft: 2,
+        width: 105,
+        ':focus': {
+            borderColor: BaseTheme.palette.primary.dark,
+        },
+    }));
+
+    return (
+        <>
+            <PrimaryButton
+                data-testid="feedback-button"
+                onClick={() => setIsOpen(true)}
+                sx={{
+                    borderRadius: '20px 20px 0px 0px',
+                    position: 'fixed',
+                    bottom: (theme: Theme) => theme.spacing(10),
+                    right: (theme: Theme) => theme.spacing(-7),
+                    transform: 'rotate(-90deg)',
+                }}
+            >
+                <ModeCommentIcon fontSize="small" sx={{ marginRight: 1 }} /> Feedback
+            </PrimaryButton>
+            <Modal aria-labelledby="modal-title" open={isOpen} onClose={() => handleClose()}>
+                <Grid
+                    container
+                    direction="row"
+                    justifyContent="space-around"
+                    alignItems="flex-start"
+                    sx={{ ...modalStyle, paddingX: 2, width: 370 }}
+                    rowSpacing={2}
+                >
+                    <ConditionalComponent condition={isSubmitted}>
```

Comment:
This here would be a perfect use case for the If Then Else in react-if

Reply:
Have we implemented the react-if yet?

Reply:
Implemented.